rust/datafusion/src/optimizer/filter_push_down.rs

```diff
@@ -215,13 +215,38 @@ fn issue_filters(
     push_down(&state, &plan)
 }
 
+/// converts "A AND B AND C" => [A, B, C]
+fn split_members(predicate: &Expr) -> Vec<&Expr> {
+    match predicate {
+        Expr::BinaryExpr {
+            right,
+            op: Operator::And,
+            left,
+        } => {
+            let mut a = split_members(&left);
+            a.extend(split_members(&right));
```

Comment:
I think this could be O(n*n) in complexity as expression trees are often nested like `1+(1+(1+(...` and extend just iterates over the right side.
Probably requires a pretty big expression to become a problem, but still maybe something to think about?

Reply:
Simple solution would be passing /inserting to a mutable vec instead

Reply:
Thanks @Dandandan . This is likely a small performance issue given that it does not affect execution, but I addressed it anyways.